### PR TITLE
CORE-12165 prep feature branch for merge

### PR DESCRIPTION
**Pull request by dickon**
- Remove state manager that will be reviewed in a separate PR.
- Fix tests.
- Use data class for key derivation parameters.

### CODE REVIEW

File: components/crypto/crypto-service-impl/src/test/kotlin/net/corda/crypto/service/impl/infra/TestServicesFactory.kt [hunk 1]
```diff
@@ -38,7 +38,7 @@ import kotlin.test.assertEquals
  * Provide instances of high level crypto services, with no database underneath, for
  * use for integration test cases that don't involve in-memory databases.
  */
-class TestServicesFactory {
+class     TestServicesFactory {
```
Comment (anien) [inline components/crypto/crypto-service-impl/src/test/kotlin/net/corda/crypto/service/impl/infra/TestServicesFactory.kt]: We should revert this white space change.

Comment (dickon) [reply components/crypto/crypto-service-impl/src/test/kotlin/net/corda/crypto/service/impl/infra/TestServicesFactory.kt]: done

**Review by anien — DISMISSED** [review]
LGTM!

**Review by anien — APPROVED** [review]
LGTM.

**Pull request merged**